docs/development/releasing.rst

```diff
@@ -1,233 +1,386 @@
 
-      $ git add .mailmap
-      $ git add docs/credits.rst
-      $ git commit -m "Updated list of contributors and .mailmap file"
+You should then run the script in the root of the astropy repository as follows::
 
-   This last commit should be forward-ported to the ``main`` branch.
+   xonsh generate_releaserst.xsh 4.3 v5.0.dev \
+                                 --project-name=astropy \
+                                 --pretty-project-name=astropy \
+                                 --pat=<a GitHub personal access token>
 
-#. Push the release branch back to GitHub, e.g.::
+The first argument should be the last major version (before any bug fix
+releases, while the second argument should be the ``.dev`` tag that was just
+after the branching of the last major version. Finally, you will need a
+GitHub personal access token with default permissions (no scopes selected).
+
+The output will look similar to::
+
+   This release of astropy contains 2573 commits in 163 merged pull requests
+   closing 104 issues from 98 people, 50 of which are first-time contributors
+   to astropy.
+
+   * 2573 commits have been added since 4.3
+   * 104 issues have been closed since 4.3
+   * 163 pull requests have been merged since 4.3
+   * 98 people have contributed since 4.3
+   * 50 of which are new contributors
+
+   The people who have contributed to the code for this release are:
+
+   - Name 1 *
+   - Name 2 *
+   - Name 3
+
+At this point, you will likely need to update the Astropy ``.mailmap`` file,
+which maps contributor emails to names, as there are often contributors who
+are not careful about using the same e-mail address for every commit, meaning
+that they appear multiple times in the contributor list above, sometimes with
+different spelling, and sometimes you may also just see their GitHub username
+with no full name.
+
+The easiest way to get a full list of contributors and email addresses is
+to do::
+
+   git shortlog -n -s -e
+
+Edit the ``.mailmap`` file to add entries for new email addresses for already
+known contributors (matched to the appropriate canonical name/email address).
+You can also try and investigate users with no name to see if you can determine
+their full name from other sources - if you do, add a new entry for them in
+the ``.mailmap`` file. Once you have done this, you can re-run the
+``generate_releaserst.xsh`` script (you will likely need to iterate a few times).
+Once you are happy with the output, copy it into the 'What's new' page for
+the current release and commit this. E.g., ::
+
+   $ git add docs/whatsnew/5.0.rst
+   $ git commit -m "Added contributor statistics and names"
+
+Update the ``docs/credits.rst`` file to include any new contributors from the
+above step, and commit this and the ``.mailmap`` changes::
+
+   $ git add .mailmap
+   $ git add docs/credits.rst
+   $ git commit -m "Updated list of contributors and .mailmap file"
+
+This last commit should be forward-ported to the ``main`` branch.
```

Comment:
Would be better to do the mailmap+credits.rst PR on the main branch and then backport.

Reply:
Done